### ISSUE TITLE
optimize targeting: make turret aware of itself/obstacles

### ISSUE THREAD

**Issue by 4rz0** [OpenModularTurretsTeam/OpenModularTurrets#119]
I placed a turret on top of my pretty flat house, and set it to target below it.
Now, if a mob spawns directly next to the building or behind something else, the turret tries to shoot it, but fails, *continuously shooting into the roof* of the building until either the energy or ammo is depleted.
This makes the turrets pretty useless, since they lock onto mobs they can never hit.
This also happens if you have a long range turret, as they try to shoot at mobs behind landscape structures like hills, if they are in range.

**Some examples:**
[Screenshot 1](https://www.dropbox.com/s/umgkp8o89wf0jau/2015-06-06_19-55-03.png?dl=0)
As you can see, even I can hardly see the mob, the turret is far right behind the turret base, but still tries to shoot trough the building sturcture.
[Screenshot 2](https://www.dropbox.com/s/vsb36epuv46cuqo/2015-06-06_20-07-48.png?dl=0)
Here, you can see a few mobs on the radar, the turret tries to shoot the one behind the vanilla building, completely ignoring the mobs it could see a little bit further away.
Another thing you'll notice here is that it shoots trough its base.

**TL;DR:** add line-of-sight check before/while turret firing.

**Comment by Keridos** [comment]
Does this also happen with the newest dev builds?

**Comment by 4rz0** [comment]
Sorry, I forgot to mention that I'm using Version 1.3.8 of OpenMouldarTurrets.
I can give the dev build a try if I get back home in a few hours.

**Comment by Keridos** [comment]
https://www.ironhide.de/jenkins/job/OpenModularTurrets/ is where those are located, remember to backup your world before adding it in, as it may break something.

**Comment by 4rz0** [comment]
Sure thing, will report back after giving it a try.

**Comment by Keridos** [comment]
We did change a lot since 1.3.8 so I am pretty sure this should be fixed. Gonna take a bit longer though until we officially release it. Still not completely done but I guess you will like that update.

**Comment by Poenjabiesous** [comment]
This was fixed with #90, #83, #80, #79, #76, #75 and #74, and the issue is no longer valid. Keep an eye out for the 2.0.0 release.
